client/lib/cart-values/cart-items.js

```diff
@@ -404,7 +410,11 @@ function themeItem( themeSlug, source ) {
  * @returns {Object} the new item as `CartItemValue` object
  */
 function domainRegistration( properties ) {
-	return assign( domainItem( properties.productSlug, properties.domain, properties.source ), { is_domain_registration: true } );
+	return assign( domainItem( properties.productSlug, properties.domain, properties.source ),
+		{
+			is_domain_registration: true,
+			...( properties.extra ? { extra: properties.extra } : {} ),
```

Comment:
This is actually not needed. Because the shopping cart is magical ;)
OK, seriously - the way I understand it, we create the item on the frontend, send it for validation to the backend. That's were the proper `extra` is written, sometimes additional products are added to the cart if needed, and the new shopping cart is returned. And _that_ one is the "canonical" one. That's how the `registrar` field is used right now.